homeassistant/helpers/script.py

```diff
@@ -1239,30 +1239,20 @@ async def _async_run_script(self, script: Script) -> None:
 class _QueuedScriptRun(_ScriptRun):
     """Manage queued Script sequence run."""
 
-    lock_acquired = False
-
     async def async_run(self) -> None:
         """Run script."""
         # Wait for previous run, if any, to finish by attempting to acquire the script's
         # shared lock. At the same time monitor if we've been told to stop.
         try:
-            async with async_interrupt.interrupt(self._stop, ScriptStoppedError, None):
-                await self._script._queue_lck.acquire()  # noqa: SLF001
+            async with (
+                self._script._queue_lck,  # noqa: SLF001
+                async_interrupt.interrupt(self._stop, ScriptStoppedError, None),
+            ):
+                await super().async_run()
         except ScriptStoppedError as ex:
             # If we've been told to stop, then just finish up.
-            self._finish()
```

Comment:
`await super().async_run()` has a call to `self._finish()` in its `finally:` so we don't need it here